version-published/snap_info_utility.py

```diff
@@ -78,14 +79,16 @@ def get_previous_tag(base_version: str, repo_path: str):
     ).splitlines()
 
     # Filter the list of tags to only include the ones that start with 'v'
-    tags = [t for t in tags if t.startswith("v")]
+    tags = [t[1:] for t in tags if t.startswith("v")]
 
     # Get the previous tag corresponding to the base version. We have to do it
     # this way because the tags are only created once the version is published.
     # For example, 4.0.0.dev333 will use the previous tag v3.3.0 to calculate
     # the offset, not v4.0.0. The versions after 4.0.0 will use v4.0.0.
     try:
-        return next(t for t in tags if Version(t) < Version(base_version))
+        return next(
+            f"v{t}" for t in tags if Version(t) < Version(base_version)
```

Comment:
I find the handling of the `v` prefix a little hard to follow. The `v` is removed when the list of tags is filtered and then re-introduced in this line. Why is that better than e.g. `next(t[1:] for t in tags if Version(t) < Version(base_version))`

Reply:
It's basically the same. The `pakaging.version` module handled both versions with v or without v in front the same way, but `distutils.version` does not handle that.

I decided to create the list of tags without the trailing "v" because in the output of `get_version_and_offset` returns the base version without the "v", so I the `tags` list with the same format.

If you think it is more confusing that way, I can trim the starting "v" in the iterator.